### ISSUE TITLE
Get.until not Working

### ISSUE THREAD

**Issue by skylastn** [Aniketkhote/refreshed#31]
Get.until((value) {
          Get.log(
            'run until : ${Get.currentRoute} == /example (${Get.currentRoute == '/example'})',
          );
          return Get.currentRoute == '/example';
        });
i try to use Get.until , but nothing happened.

**Comment by Aniketkhote** [comment]
Check with latest flutter and refreshed version

**Comment by skylastn** [comment]
Still don't work bro
the logs say but not release getx controller
[Refreshed] GetInformationParser: Parsing route location: /admin
2
[Refreshed] Restoring route information for route: /admin
u can test it when using back button from browser not the apps

**Comment by Gaurav-CareMonitor** [comment]
@Aniketkhote ?

**Comment by Aniketkhote** [comment]
Hey! First off, I'm sorry this issue has been pending for so long. I know it was raised almost a year ago, and by now it might not even be relevant to your current setup, but I still consider it a serious problem that needs to be addressed properly.

From your description, it looks like Get.until() isn’t working as expected when triggered by a button, and Get.offAllNamed() isn’t fully cleaning up the controllers, which is concerning. I’ll be taking a closer look at this behavior soon, especially around route cleanup and controller lifecycle.

In the meantime, if you’ve found any workaround or solution on your side, please do share! It would really help guide the fix and ensure it works across more real-world use cases.

Thanks again for your patience and for pointing this out 🙏

**Comment by ggdream** [comment]
@Aniketkhote Hello, my application project also has a problem here. I simply debugged this package through my business logic. The final location is here。

This can be understood by combining other people's code examples as the second to last page being moved to the last to last position, which causes the order of the pages to be messed up.

https://github.com/Aniketkhote/refreshed/blob/17d9bfd6804ffdef7112704879d54a9a0e2f0076/lib/get_navigation/src/routes/get_router_delegate.dart#L759-L760

**Comment by ggdream** [comment]
> [@Aniketkhote](https://github.com/Aniketkhote) Hello, my application project also has a problem here. I simply debugged this package through my business logic. The final location is here。
> 
> This can be understood by combining other people's code examples as the second to last page being moved to the last to last position, which causes the order of the pages to be messed up.
> 
> [refreshed/lib/get_navigation/src/routes/get_router_delegate.dart](https://github.com/Aniketkhote/refreshed/blob/17d9bfd6804ffdef7112704879d54a9a0e2f0076/lib/get_navigation/src/routes/get_router_delegate.dart#L759-L760)
> 
> Lines 759 to 760 in [17d9bfd](/Aniketkhote/refreshed/commit/17d9bfd6804ffdef7112704879d54a9a0e2f0076)
> 
>  _activePages.remove(existingPage); 
>  _activePages.add(res);

Obviously, it is impossible to determine whether the user pressed the back button through the information provided by Flutter. So the best way I can think of is to compare it through the routing stack. If it already exists in the stack, pop it to that position. If not, push it. Of course, this implementation is somewhat inconsistent with the usual web behavior, but this is the best solution for the current situation. At least it will not cause routing confusion.

By the way, whether the button returns can be indirectly realized by monitoring the pop state through the package:web package, but this implementation is a bit irregular.

```dart
  @override
  SynchronousFuture<RouteDecoder> parseRouteInformation(
    RouteInformation routeInformation,
  )
```
